pkg/controller/daemon/daemon_controller.go

```diff
@@ -1236,6 +1236,15 @@ func (dsc *DaemonSetsController) syncDaemonSet(ctx context.Context, key string)
 		return dsc.updateDaemonSetStatus(ctx, ds, nodeList, hash, false)
 	}
 
+	defer func() {
+		statusErr := dsc.updateDaemonSetStatus(ctx, ds, nodeList, hash, true)
+		if statusErr != nil && syncErr != nil {
+			klog.V(4).ErrorS(err, "Failed to update status", "daemonSet", klog.KObj(ds))
+		} else if syncErr == nil {
```

Comment:
maybe we should test that we are getting the correct error in  both cases?

Reply:
added a test.